File: clean.py
import os
import json

data = None

with open('./here_data.json') as f:
    data = json.load(f)

final = []
last_lat = 0
last_long = 0

print(final)

for piece in data:
    if not (abs(piece["position"]["lat"] - last_lat) < .0002 and
            abs(piece["position"]["lng"] - last_long) < .0002):
        last_lat = piece["position"]["lat"]
        last_long = piece["position"]["lng"]

        temp = {
            "lat": piece["position"]["lat"],
            "long": piece["position"]["lng"],
            "timestamp": piece["timestamp"]
        }

        final.append(temp)


with open('./position_data_cleaned.json', 'w+') as f:
    json.dump(final, f)

File: clean.js
/**
 * Function to clean the json object to only contain lat/long values a certain
 * distance apart
 * @param {*} json_list - a list of json objects to be cleaned
 * @return - the cleaned json object 
 */
function clean_lat_longs (json_list) {
  last_lat = 0
  last_long = 0

  // accumulator for cleaned location values
  final = [];

  for (index in json_list) {
    json_obj = json_list[index]
    if (!((Math.abs(json_obj.position.lat - last_lat) < .0002) &&
      (Math.abs(json_obj.position.lng - last_long) < .0002))) {
      last_lat = json_obj.position.lat
      last_long = json_obj.position.lng

      temp = {
        'lat': last_lat,
        'long': last_long,
        'timestamp': json_obj.timestamp
      }
      final.push(temp)
    }
  }

  return final
}

// exporting clean_lat_longs to use in api.js
module.exports = {
  clean_lat_longs: clean_lat_longs
}

console.log(clean_lat_longs(data))
